libs/payments/stripe/src/lib/accountCustomer/accountCustomer.repository.ts

```diff
@@ -0,0 +1,94 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {
+  AccountCustomerUpdate,
+  AccountDatabase,
+  NewAccountCustomer,
+} from '@fxa/shared/db/mysql/account';
+
+import {
+  AccountCustomerManagerError,
+  AccountCustomerNotUpdatedError,
+} from './accountCustomer.error';
+
+export async function createAccountCustomer(
+  db: AccountDatabase,
+  accountCustomer: NewAccountCustomer
+) {
+  await db
+    .insertInto('accountCustomers')
+    .values(accountCustomer)
+    .executeTakeFirstOrThrow();
+  return getAccountCustomerByUid(db, accountCustomer.uid);
+}
+
+/**
+ * Locate an accountCustomer record by fxa uid
+ * @returns AccountCustomer | undefined
+ */
+export async function getAccountCustomerByUid(
+  db: AccountDatabase,
+  uid: Buffer
+) {
+  return db
+    .selectFrom('accountCustomers')
+    .where('uid', '=', uid)
+    .selectAll()
+    .executeTakeFirstOrThrow();
+}
+
+/**
+ * Attempts to update an accountCustomer record by fxa user id
+ * Returns the number of affected rows
+ *
+ * @returns number of affected rows
+ */
+export async function updateAccountCustomer(
+  db: AccountDatabase,
+  uid: Buffer,
+  update: Omit<AccountCustomerUpdate, 'uid'>
+): Promise<boolean> {
+  // Re-build properties for type-safety (Typescript allows wider type to be applied to narrower object type)
+  const _update: Omit<AccountCustomerUpdate, 'uid'> = {};
+  if (update.stripeCustomerId !== undefined)
+    _update.stripeCustomerId = update.stripeCustomerId;
+  if (update.createdAt !== undefined) _update.createdAt = update.createdAt;
+  if (update.updatedAt !== undefined) _update.updatedAt = update.updatedAt;
```

Comment:
Interestingly it looks like the type for AccountCustomerUpdate doesn't allow these arguments to be undefined